src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesigner.cs

```diff
@@ -70,11 +72,14 @@ public override bool RunTask ()
 
 			// Create our capitalization maps so we can support mixed case resources
 			foreach (var item in Resources) {
-				if (!item.ItemSpec.StartsWith (ResourceDirectory))
+				var path = Path.GetFullPath (item.ItemSpec);
+				if (!path.StartsWith (ResourceDirectory))
 					continue;
 
-				var name = item.ItemSpec.Substring (ResourceDirectory.Length);
+				var name = path.Substring (ResourceDirectory.Length).TrimStart ('/', '\\');
```

Comment:
…otherwise the `.StartsWith()` will be a "culture aware" comparison, and `path.Substring(ResourceDirectory.Length)` won't remove what you expect.

For example:

```csharp
// a & b are 'é'
var a = "\u00e9";
var b = "\u0065\u0301";                              // combining chars!
a.StartsWith(b);                                     // true
a.StartsWith(b, StringComparison.OrdinalIgnoreCase); // false
```

Now it's possible I'm overthinking this, because on Windows (at least) filesystem names are normalized, so there shouldn't be a normalization mismatch between `ResourceDirectory` and `Resources`.  However, other platforms (Linunx, macOS+APFS) perform *no* normalization, which frankly just makes this entire scenario *scarier*.